data_sources/kafka_data_source.py

```diff
@@ -0,0 +1,61 @@
+from data_sources.data_source import DataSource
+from confluent_kafka import Consumer, KafkaError
+import time
+import json
+
+
+class KafkaDataSource(DataSource):
+    def __init__(self, kafka_endpoint, kafka_topic, s3_bucket, read_timeout_secs, batch_size = 1024):
+        super().__init__(s3_bucket)
+        self.kafka_endpoint = kafka_endpoint
+        self.kafka_topic = kafka_topic
+        self.read_timeout_secs = read_timeout_secs
+        self.batch_size = batch_size
+
+    def read_data_into_s3(self, file_size=1024 * 1024):
+        # Configure Kafka consumer
+        consumer_conf = {
+            'bootstrap.servers': self.kafka_endpoint,
+            'group.id': 'tdsd-group',
+            'auto.offset.reset': 'earliest'
+        }
+        consumer = Consumer(consumer_conf)
+        consumer.subscribe([self.kafka_topic])
+
+        # Initialize variables
+        current_file_size = 0
+        current_file_data = []
+
+        start_time = time.time()
+        try:
+            while (time.time() - start_time) < self.read_timeout_secs:
+                messages = consumer.consume(self.batch_size, timeout=1.0)
+
+                if messages is None:
+                    continue
+                for message in messages:
+                    if message.error():
+                        if message.error().code() == KafkaError._PARTITION_EOF:
+                            # End of partition event
+                            break
+                        else:
+                            print(f"Error: {message.error().str()}")
+                            continue
+
+                    data = json.loads(message.value().decode("utf-8"))
```

Comment:
utf-8 is a standard for Kafka messages or should it be a variable?

Reply:
I will check, but I think its pretty standart